File: AutoIndustryMarketPlace/lib/models/quote.dart
class Quote {
  final String id;
  final String repairShopId;
  final String requestId;
  final Map<String, double> priceBreakdown;
  final double estimatedHours;
  final double totalPrice;

  Quote({
    required this.id,
    required this.repairShopId,
    required this.requestId,
    required this.priceBreakdown,
    required this.estimatedHours,
    required this.totalPrice,
  });

  factory Quote.fromJson(Map<String, dynamic> json) {
    return Quote(
      id: json['id'],
      repairShopId: json['repairShopId'],
      requestId: json['requestId'],
      priceBreakdown: Map<String, double>.from(json['priceBreakdown'] ?? {}),
      estimatedHours: json['estimatedHours'],
      totalPrice: json['totalPrice'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'requestId': requestId,
      'priceBreakdown': priceBreakdown,
      'estimatedHours': estimatedHours,
      'totalPrice': totalPrice,
    };
  }
}

File: Autoindustry-Backend/src/main/java/com/autoindustry/backend/model/Quote.java
package com.autoindustry.backend.model;

import com.fasterxml.jackson.databind.JsonNode;
import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "quotes")
@Data
public class Quote {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "repair_shop_id")
    private User repairShop;

    @Column(name = "request_id")
    private String requestId;  // Reference to MongoDB ID

    @Column(columnDefinition = "jsonb")
    private JsonNode priceBreakdown;

    private double estimatedHours;

    @Column
    private double totalPrice;
}